pkg/controller/egress/controller_test.go

```diff
@@ -766,3 +766,81 @@ func checkExternalIPPoolUsed(t *testing.T, controller *egressController, poolNam
 	})
 	assert.NoError(t, err)
 }
+
+func TestEgressStatus(t *testing.T) {
```

Comment:
After making the above change, we could just have an unit test for `updateEgressAllocatedCondition`, which just focus on validating the condition based on given egress and error.